core/controllers/email_dashboard_test.py

```diff
@@ -152,6 +184,262 @@ def setUp(self):
         self.set_admins(
             [self.SUBMITTER_USERNAME, self.NEW_SUBMITTER_USERNAME])
 
+    def test_handler_with_invalid_num_queries_to_fetch_raises_error_400(self):
+        self.login(self.SUBMITTER_EMAIL)
+
+        response = self.get_json(
+            '/emaildashboarddatahandler',
+            params={'num_queries_to_fetch': 'invalid_num_queries_to_fetch'},
+            expected_status_int=400)
+        self.assertEqual(
+            response['error'], '400 Invalid input for query results.')
+
+        self.logout()
+
+    def test_email_dashboard_data_handler(self):
+        self.login(self.SUBMITTER_EMAIL)
+
+        response = self.get_json(
+            '/emaildashboarddatahandler',
+            params={'num_queries_to_fetch': 1})
+        self.assertEqual(response['recent_queries'], [])
+
+        query_id = user_query_services.save_new_query_model(
+            self.submitter_id, inactive_in_last_n_days=10,
+            created_at_least_n_exps=5,
+            has_not_logged_in_for_n_days=30)
+
+        response = self.get_json(
+            '/emaildashboarddatahandler',
+            params={'num_queries_to_fetch': 1})
+
+        self.assertEqual(response['recent_queries'][0]['id'], query_id)
+        self.assertEqual(
+            response['recent_queries'][0]['status'],
+            feconf.USER_QUERY_STATUS_PROCESSING)
+
+        self.logout()
+
+    def test_email_dashboard_result_page_with_invalid_query_id_raises_400(self):
+        self.login(self.SUBMITTER_EMAIL)
+
+        query_id = user_query_services.save_new_query_model(
+            self.submitter_id, inactive_in_last_n_days=10,
+            created_at_least_n_exps=5,
+            has_not_logged_in_for_n_days=30)
+
+        job_id = user_query_jobs_one_off.UserQueryOneOffJob.create_new()
+        params = {'query_id': query_id}
+        user_query_jobs_one_off.UserQueryOneOffJob.enqueue(
+            job_id, additional_job_params=params)
+
+        # Complete execution of query.
+        self.assertEqual(
+            self.count_jobs_in_taskqueue(
+                taskqueue_services.QUEUE_NAME_ONE_OFF_JOBS), 1)
```

Comment:
Nit: Please put expected value on new line to help readers (here and elsewhere).

Reply:
Done